internal/state/module.go

```diff
@@ -397,7 +427,14 @@ func (s *ModuleStore) UpdateParsedFiles(path string, pFiles map[string]*hcl.File
 		return err
 	}
 
-	mod.ParsedFiles = pFiles
+	langID, _ := lsp.LanguageIDFromPath(path)
```

Comment:
I initially thought we would create a new dedicated operation such as `ParseVariables` in `internal/terraform/module/module_ops.go` and relatedly create a new dedicated method here, e.g.

```go
func (s *ModuleStore) UpdateVariableFiles(path string, pFiles map[string]*hcl.File, pErr error) error
```
at which point the separation would be clearer and only _locally_ based on file extensions inside the operation code - no need to care about language IDs there.

If this is _not_ what you plan to do then I'd probably reconsider the whole idea of storing the files separately, because that makes it more difficult here and in a few other places.